akka-stream/src/main/scala/akka/stream/impl/io/TcpStages.scala

```diff
@@ -272,14 +274,18 @@ private[stream] object TcpConnectionStage {
 
       override def onUpstreamFailure(ex: Throwable): Unit = {
         if (connection != null) {
-          if (interpreter.log.isDebugEnabled) {
-            interpreter.log.debug(
-              "Aborting tcp connection because of upstream failure: {}\n{}",
-              ex.getMessage,
-              ex.getStackTrace.mkString("\n"))
-          }
           connection ! Abort
-        } else failStage(ex)
+        } else {
+          import scala.concurrent.duration._
+          val knownIdleTimeoutPrefix = "No elements passed in the last "
```

Comment:
The string matching is a short hack. Wanted feedback mostly on the exception (lines 283). 
However allowing the timeouts to customise the exception type is a good idea, I'll try that 👍